types/prosemirror-model/index.d.ts

```diff
@@ -1,259 +1,1290 @@
+  parseSlice(dom: Node, options?: ParseOptions): Slice;
+  /**
+   * Construct a DOM parser using the parsing rules listed in a
+   * schema's [node specs](#model.NodeSpec.parseDOM), reordered by
+   * [priority](#model.ParseRule.priority).
+   */
   static fromSchema(schema: Schema): DOMParser;
 }
+/**
+ * A mark is a piece of information that can be attached to a node,
+ * such as it being emphasized, in code font, or a link. It has a type
+ * and optionally a set of attributes that provide further information
+ * (such as the target of the link). Marks are created through a
+ * `Schema`, which controls which types exist and which
+ * attributes they have.
+ */
 export class Mark {
+  /**
+   * The type of this mark.
+   */
   type: MarkType;
-  attrs: AnyObject;
+  /**
+   * The attributes associated with this mark.
+   */
+  attrs: { [key: string]: any };
+  /**
+   * Given a set of marks, create a new set which contains this one as
+   * well, in the right position. If this mark is already in the set,
+   * the set itself is returned. If any marks that are set to be
+   * [exclusive](#model.MarkSpec.excludes) with this mark are present,
+   * those are replaced by this one.
+   */
   addToSet(set: Mark[]): Mark[];
+  /**
+   * Remove this mark from the given set, returning a new set. If this
+   * mark is not in the set, the set itself is returned.
+   */
   removeFromSet(set: Mark[]): Mark[];
+  /**
+   * Test whether this mark is in the given set of marks.
+   */
   isInSet(set: Mark[]): boolean;
+  /**
+   * Test whether this mark has the same type and attributes as
+   * another mark.
+   */
   eq(other: Mark): boolean;
-  toJSON(): AnyObject;
-  static fromJSON(schema: Schema, json: AnyObject): Mark;
+  /**
+   * Convert this mark to a JSON-serializeable representation.
+   */
+  toJSON(): { [key: string]: any };
+  static fromJSON(schema: Schema, json: { [key: string]: any }): Mark;
+  /**
+   * Test whether two sets of marks are identical.
+   */
   static sameSet(a: Mark[], b: Mark[]): boolean;
+  /**
+   * Create a properly sorted mark set from null, a single mark, or an
+   * unsorted array of marks.
+   */
   static setFrom(marks?: Mark | Mark[]): Mark[];
+  /**
+   * The empty set of marks.
+   */
   static none: Mark[];
 }
-export class Node {
+/**
+ * This class represents a node in the tree that makes up a
+ * ProseMirror document. So a document is an instance of `Node`, with
+ * children that are also instances of `Node`.
+ *
+ * Nodes are persistent data structures. Instead of changing them, you
+ * create new ones with the content you want. Old ones keep pointing
+ * at the old document shape. This is made cheaper by sharing
+ * structure between the old and new data as much as possible, which a
+ * tree shape like this (without back pointers) makes easy.
+ *
+ * **Do not** directly mutate the properties of a `Node` object. See
+ * [the guide](/docs/guide/#doc) for more information.
+ */
+declare class ProsemirrorNode {
```

Comment:
Did you hit any gotchas with going this route?

Reply:
No.